src/dasi/core/FileReadHandle.cc

```diff
@@ -0,0 +1,49 @@
+
+#include "dasi/core/FileReadHandle.h"
+
+#include <iostream>
+
+namespace dasi::core {
+
+FileReadHandle::FileReadHandle(const std::filesystem::path& path, long long offset) :
```

Comment:
This is an odd mix. I would expect either to open a file, or to open a file with an offset and length (i.e. opening a subset of a file). But not having a length option here is a bit odd. Implies that the handle will always read to the end of the file.

Reply:
You could have a separate handle type. You could have a separate constructor. Or you could have length=-1 reads until the end of the file (with that as the default). I don't feel strongly.